src/messaging/tests/TestMessagingLayer.cpp

```diff
@@ -142,6 +142,26 @@ void CheckExchangeOutgoingMessagesFail(nlTestSuite * inSuite, void * inContext)
     ec->Close();
 }
 
+/**
+ * Tests that the Exchange Context APIs do not crash if delayed calls are made after the Exchange Context is
+ * closed.
+ */
+void CheckExchangeContextDoesNotCrashWhenDelayedCallsOccurAfterClose(nlTestSuite * inSuite, void * inContext)
```

Comment:
Also, this test does not compile, and if it compiled would not get run.  Please make sure to verify that tests fail without the fix and pass with it...